test/test_asarray.py

```diff
@@ -0,0 +1,423 @@
+#   characteristics:
+#       1. is a Tensor
+#       2. is a DLPack capsule
+#       3. implements the Python Buffer protocol
+#       4. is an arbitrary list
+#   The implementation itself is based on the Python Array API:
+#   https://data-apis.org/array-api/latest/API_specification/creation_functions.html
+class TestAsArray(common.TestCase):
+    def _check(self, original, cvt=lambda t: t, is_alias=True, same_dtype=True, same_device=True, **kwargs):
+        """Check the output of 'asarray', given its input and assertion informations.
+
+        Besides calling 'asarray' itself, this function does 4 different checks:
+            1. Whether the result is aliased or not, depending on 'is_alias'
+            2. Whether the result has the expected dtype and elements
+            3. Whether the result lives in the expected device
+            4. Whether the result has its 'requires_grad' set or not
+        """
+        result = torch.asarray(cvt(original), **kwargs)
+        original_tensor = gettensorlike(original)
+        self.assertTrue(isinstance(result, torch.Tensor))
+
+        # 1. The storage pointers should be equal only if 'is_alias' is set
+        if is_alias:
+            self.assertEqual(result.data_ptr(), getaddr(original))
+        else:
+            self.assertNotEqual(result.data_ptr(), getaddr(original))
+
+        # 2. Comparison of the elements only takes place if the original
+        # sequence and the resulting tensor have the same data type
+        if same_dtype:
+            self.assertEqual(original_tensor, result)
+        else:
+            dtype = kwargs.get("dtype", torch.get_default_dtype())
+            self.assertEqual(original_tensor.shape, result.shape)
+            self.assertEqual(dtype, result.dtype)
+
+
+        # 3. Given the specified target device, we first check whether
+        # its type is the same, and then if its index is the same (if it
+        # is not None)
+        if same_device:
+            device = getdevice(original)
+        else:
+            device = torch.device(kwargs.get("device", "cpu"))
+
+        # Compare the target device type, and its index
+        self.assertEqual(device.type, result.device.type)
+        if device.index is not None:
+            self.assertEqual(device.index, result.device.index)
+
+        # 4. By default, 'requires_grad' is unset
+        self.assertEqual(result.requires_grad, kwargs.get("requires_grad", False))
+
+    def _may_compute_grad(self, dtype):
+        return dtype.is_floating_point or dtype.is_complex
+
+    # Skipping 'meta' devices, since there's no point in comparing their
+    # data pointer (which is basically the point here), since they all
+    # return 0.
+    @skipMeta
+    @dtypes(*get_all_dtypes())
+    def test_alias_from_tensor(self, device, dtype):
+        # DLpack does not explicitly support bool
+        # It does it through uint8 type
+        if dtype is torch.bool:
```

Comment:
Nit: skip the test instead of just returning (here and in the other tests)

Reply:
Also: while this is true, DLPack isn't tested here, is it? So maybe this skip is not needed

Reply:
Ah, you are right. I had completely missed that.